### DISCUSSION TITLE
Nanopore sequencing - Drastic differences in results between v4.3.2 and v4.4.1.

### DISCUSSION THREAD

**Discussion by jingquanlim** [milaboratory/mixcr#1264] [General]
Hi all, 

I have gotten the attached "barcdoe23.fastq.gz" file from Nanopore sequencing of TCR amplicons. A total of ~18k reads were generated and was ran on Mixcr v4.3.2 (preset: ont-rna-seq-vdj-full-length) and Mixcr v4.4.1 (preset: ont-rna-seq).

The data was validated with Smartseq2; CDR3 was found to be "CASSLGKGNQPQHF". v4.3.2 returned this as the top hit with 553 reads supporting this result. However, v4.4.1 failed at the 'assemble' step with

`error msg : "picocli.CommandLine$ExecutionException: Error while running command assemble java.lang.ArrayIndexOutOfBoundsException: Index 90 out of bounds for length 70".`

Additional param "--assemble-clonotypes-by 'CDR3+FR4'" was added to the v4.4.1 run and the top hit showed only 9 reads supporting this result. I guess a lot more QC metrics are now considered in v4.4.1 than in v.4.3.2. Since the change is pretty drastic, maybe we can discuss how we should mitigate this deficit in v4.4.1 in handling ONT reads better. 

Thanks!
JQ

[barcode23.fastq.gz](https://github.com/milaboratory/mixcr/files/12190057/barcode23.fastq.gz)

**Comment by Alex-Davydov** [comment]
Hi,
Try the latest version of MiXCR (4.4.2), the ont preset should work without any additional parameters.

**Comment by jingquanlim** [comment]
Thanks much Alex. Will try it out soon.

JQ

On Wed, 2 Aug 2023 at 09:35, Alex-Davydov ***@***.***> wrote:

> Hi,
> Try the latest version of MiXCR (4.4.2), the ont preset should work
> without any additional parameters.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/milaboratory/mixcr/discussions/1264#discussioncomment-6611258>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOEMXXYXWVCP75SBPWKKO3XTGVETANCNFSM6AAAAAA22665NI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 

Regards,
Jing Quan

**Comment by mizraelson** [comment]
Hi Jing,
Just wanted to follow up and ask if everything is running smoothly with the latest MiXCR version. Is there anything I can help you with?

Sincerely,
Mark

**Comment by jingquanlim** [comment]
Hi Mark,

4.4.2 works fine and Dillon, our intern, has managed to reproduced the
results from 4.3.2 too. So I guess its working fine now.

Thanks for checking back and your help on this issue.

JQ

On Wed, 9 Aug 2023 at 17:30, mizraelson ***@***.***> wrote:

> Hi Jing,
> Just wanted to follow up and ask if everything is running smoothly with
> the latest MiXCR version. Is there anything I can help you with?
>
> Sincerely,
> Mark
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/milaboratory/mixcr/discussions/1264#discussioncomment-6678361>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOEMXU6SAGGX6RA56TBA73XUNKCRANCNFSM6AAAAAA22665NI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 

Regards,
Jing Quan

**Reply by mizraelson** [reply]
Great!
